### PR TITLE
Prevent looking up a parent if there is none

### LINKED ISSUE #8436: SmallRyeFaultToleranceProcessor#hasFTAnnotations(line 222) - crashes on java.lang.Object Beans -> no Super Class
**Describe the bug**
Version 1.3.1. In case a cdi bean does not have a super class -> java.lang.Object, we are getting:
```
java.lang.RuntimeException: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.smallrye.faulttolerance.deployment.SmallRyeFaultToleranceProcessor#validateFaultToleranceAnnotations threw an exception: java.lang.IllegalArgumentException: Cannot lookup class, provided DotName was null.
	at io.quarkus.arc.processor.IndexClassLookupUtils.getClassByName(IndexClassLookupUtils.java:41)
	at io.quarkus.arc.processor.BeanArchives$IndexWrapper.getClassByName(BeanArchives.java:103)
	at io.quarkus.smallrye.faulttolerance.deployment.SmallRyeFaultToleranceProcessor.hasFTAnnotations(SmallRyeFaultToleranceProcessor.java:224)
	at io.quarkus.smallrye.faulttolerance.deployment.SmallRyeFaultToleranceProcessor.validateFaultToleranceAnnotations(SmallRyeFaultToleranceProcessor.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
```
 
suggestion would be to add at line 222 in SmallRyeFaultToleranceProcessor:
```
if(parentClassName == null) return false;
```
for instance the method hasFTAnnotaions would look like that:

```
private boolean hasFTAnnotations(IndexView index, AnnotationStore annotationStore, ClassInfo info) {
        // first check annotations on type
        if (annotationStore.hasAnyAnnotation(info, FT_ANNOTATIONS)) {
            return true;
        }

        // then check on the methods
        for (MethodInfo method : info.methods()) {
            if (annotationStore.hasAnyAnnotation(method, FT_ANNOTATIONS)) {
                return true;
            }
        }

        // then check on the parent
        DotName parentClassName = info.superName();
        if(parentClassName == null){
            return false;
        }

        ClassInfo parentClassInfo = index.getClassByName(parentClassName);
        if (parentClassName.equals(DotNames.OBJECT) || parentClassInfo == null) {
            return false; //no more parents
        }

        return hasFTAnnotations(index, annotationStore, parentClassInfo);
    }
```

**Expected behavior**
no exception

**To Reproduce**
1.  Use myfaces 2.3-next-M2 (download from github and build it) with quarkus 1.3.1.Final.
2. Not able to run QuakrusTests or start the quarkus server

https://github.com/apache/myfaces
e.g. add those dependencies

```
       <dependency>
            <groupId>org.apache.myfaces.core.extensions.quarkus</groupId>
            <artifactId>myfaces-quarkus-deployment</artifactId>
            <version>2.3-next-SNAPSHOT</version>
        </dependency>

        <dependency>
            <groupId>org.apache.myfaces.core.extensions.quarkus</groupId>
            <artifactId>myfaces-quarkus-runtime</artifactId>
            <version>2.3-next-SNAPSHOT</version>
        </dependency>

        <dependency>
            <groupId>org.apache.myfaces.core</groupId>
            <artifactId>myfaces-api</artifactId>
            <version>2.3-next-SNAPSHOT</version>
        </dependency>

        <dependency>
            <groupId>org.apache.myfaces.core</groupId>
            <artifactId>myfaces-impl</artifactId>
            <version>2.3-next-SNAPSHOT</version>
        </dependency>
```

### PR DESCRIPTION
**Pull request by jmartisk**
Fixes #8436

### CODE REVIEW

**Comment by Ladicek** [comment]
Care to add a test? :-)

File: extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java [hunk 1]
```diff
@@ -225,11 +225,11 @@ private boolean hasFTAnnotations(IndexView index, AnnotationStore annotationStor
...
         // then check on the parent
         DotName parentClassName = info.superName();
-        ClassInfo parentClassInfo = index.getClassByName(parentClassName);
-        if (parentClassName.equals(DotNames.OBJECT) || parentClassInfo == null) {
-            return false; //no more parents
+        if (parentClassName == null || parentClassName.equals(DotNames.OBJECT)) {
+            return false;
         }
 
+        ClassInfo parentClassInfo = index.getClassByName(parentClassName);
```
Comment (gsmet) [inline extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java:232]: I think you should still test for a null return value here.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java (hunk 1 above)
Comment (jmartisk) [reply extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java:231]: Ok, we're never safe enough

**Comment by jmartisk** [comment]
@Ladicek I added a very primitive test, please have a look if this is what you envisioned ;)

File: extensions/smallrye-fault-tolerance/deployment/src/test/java/io/quarkus/smallrye/faulttolerance/test/ObjectBeanTest.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import javax.enterprise.inject.Produces;
+import javax.inject.Named;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+/**
+ * Verify that Fault Tolerance does not break if there is a CDI bean without a superclass
```
Comment (Ladicek) [inline extensions/smallrye-fault-tolerance/deployment/src/test/java/io/quarkus/smallrye/faulttolerance/test/ObjectBeanTest.java:12]: There's no FT in the test at all?

Comment (Ladicek) [reply extensions/smallrye-fault-tolerance/deployment/src/test/java/io/quarkus/smallrye/faulttolerance/test/ObjectBeanTest.java:12]: I guess it's about the `object` being processed by the `FTProcessor`, right? Gotcha. It looks weird, but it's probably enough.

**Review by gsmet — APPROVED** [review]

File: extensions/smallrye-fault-tolerance/deployment/src/test/java/io/quarkus/smallrye/faulttolerance/test/ObjectBeanTest.java (hunk 2 above)
Comment (jmartisk) [reply extensions/smallrye-fault-tolerance/deployment/src/test/java/io/quarkus/smallrye/faulttolerance/test/ObjectBeanTest.java:12]: Yep, this is enough to reproduce the original issue if I deliberately remove the fix

Comment (Ladicek) [reply extensions/smallrye-fault-tolerance/deployment/src/test/java/io/quarkus/smallrye/faulttolerance/test/ObjectBeanTest.java:12]: Nice, thanks!

**Pull request merged**